h2o-core/src/main/java/hex/AUC2.java

```diff
@@ -278,6 +269,50 @@ private double compute_auc() {
     return area/_p/_n;
   }
 
+  /**
+   * Compute the Area under Precision-Recall Curves using TPs and FPs.
+   * TPs and FPs are monotonically increasing.
+   * Calulation inspired by XGBoost implementation:
+   * https://github.com/dmlc/xgboost/blob/master/src/metric/rank_metric.cc#L566-L591
+   * @return
+   */
+  public double pr_auc() {
+    if (isEmpty()) {
+      return Double.NaN;
+    }
+    checkRecallValidity();
+    
+    if (_fps[_nBins-1] == 0) return 1.0;
+    if (_tps[_nBins-1] == 0) return 0.0;
+    
+    double area = 0.0;
+    assert _p > 0 && _n > 0 : "AUC-PR calculation error, sum of positives and sum of negatives should be zero.";
```

Comment:
```suggestion
    assert _p > 0 && _n > 0 : "AUC-PR calculation error, sum of positives and sum of negatives should be greater than zero.";
```

Maybe we could provide the information about how many positives and negatives are there as well, but it's optional.

Reply:
Good catch, thank you!